src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -2875,184 +3030,650 @@ private void ensurePublishCapability(
 
-        private MqttWillProperties decode(
-            DirectBuffer payloadBuffer,
-            int payloadOffset,
-            int payloadLimit)
-        {
+                decodeNetworkIfNecessary(traceId);
+                cleanup(traceId, authorization);
+            }
+
+            private void onApplicationReply(
+                int msgTypeId,
+                DirectBuffer buffer,
+                int index,
+                int length)
+            {
+                switch (msgTypeId)
+                {
+                case BeginFW.TYPE_ID:
+                    final BeginFW begin = beginRO.wrap(buffer, index, index + length);
+                    onApplicationBegin(begin);
+                    break;
+                case DataFW.TYPE_ID:
+                    final DataFW data = dataRO.wrap(buffer, index, index + length);
+                    onApplicationData(data);
+                    break;
+                case EndFW.TYPE_ID:
+                    final EndFW end = endRO.wrap(buffer, index, index + length);
+                    // onApplicationEnd(end);
+                    break;
+                case AbortFW.TYPE_ID:
+                    final AbortFW abort = abortRO.wrap(buffer, index, index + length);
+                    // onApplicationAbort(abort);
+                    break;
+                }
+            }
+
+            private void onApplicationBegin(
+                BeginFW begin)
+            {
+                state = MqttState.openReply(state);
+
+                final long traceId = begin.traceId();
+                final long authorization = begin.authorization();
+
+                doApplicationWindowIfNecessary(traceId, authorization);
+            }
+
+            private void onApplicationData(
+                DataFW data)
+            {
+                final long traceId = data.traceId();
+                final int reserved = data.reserved();
+                final long authorization = data.authorization();
+                final int flags = data.flags();
+                final OctetsFW payload = data.payload();
+                final OctetsFW extension = data.extension();
+
+                replyBudget -= reserved;
+                encodeSharedBudget -= reserved;
+
+                if (replyBudget < 0)
+                {
+                    doApplicationReset(traceId, authorization);
+                    doNetworkAbort(traceId, authorization);
+                }
+                else
+                {
+                    if (payload != null)
+                    {
+                        // TODO: get topic
+                        //       decode the payload; check if will or session
+                        //       check session for existing subscriptions, subscribe client to them
+                        // final boolean retainAsPublished = subscription.retainAsPublished();
+                        // doEncodePublish(traceId, authorization, flags, subscription.id, retainAsPublished,
```

Comment:
Given that this wildcard stream is used to manage internal state only, there should be no case where we send `PUBLISH` packets directly to the client representing that internal state.

Inside this `if` block we need a method call, something like `onSessionStateUpdated(payload, extension);` to handle the processing logic.